web/srv/api_handlers.go

```diff
@@ -242,3 +246,39 @@ func (h *handler) handleApiTap(w http.ResponseWriter, req *http.Request, p httpr
 		}
 	}
 }
+
+func (h *handler) handleProfileDownload(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+	service := req.FormValue("service")
+	namespace := req.FormValue("namespace")
+
+	if service == "" || namespace == "" {
+		err := fmt.Errorf("Service and namespace must be provided to create a new profile")
+		log.Error(err.Error())
+		renderJsonError(w, err, http.StatusBadRequest)
```

Comment:
I wouldn't expect these error messages to be rendered as JSON, since this endpoint doesn't return JSON for successful responses. Can you instead do something like:

```suggestion
		http.Error(w, err.Error(), http.StatusBadRequest)
```